packages/light-apps/src/Onboarding/ImportWithJson.tsx

```diff
@@ -0,0 +1,38 @@
+// Copyright 2018-2019 @paritytech/substrate-light-ui authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { InputFile, Modal, NavLink, Stacked } from '@polkadot/ui-components';
+import { u8aToString } from '@polkadot/util';
+import React from 'react';
+import { RouteComponentProps } from 'react-router-dom';
+
+interface Props extends RouteComponentProps {}
+
+export class ImportWithJson extends React.Component<Props> {
+  private handleFileUploaded = async (file: Uint8Array) => {
+    const { history } = this.props;
+
+    const jsonString = u8aToString(file);
+
+    history.push(`/save/withJson/${jsonString}`);
```

Comment:
I think people won't be too happy if they see their JSON secret (same for recovery phrase) in the URL, even though it's in Electron and we don't show this URL (though it can be easily accessed via window.location).

You can do `history.push` with a state (see here https://reacttraining.com/react-router/core/api/history), so that these secret strings only stay in memory.